tools/protoxform/protoprint.py

```diff
@@ -204,7 +204,15 @@ def CamelCase(s):
   file_block = '\n'.join(['syntax = "proto3";\n', package_line])
 
   options = descriptor_pb2.FileOptions()
+
   options.java_outer_classname = CamelCase(os.path.basename(file_proto.name))
+  for msg in file_proto.message_type:
+    if msg.name == options.java_outer_classname:
+      # This is a workaround for Java outer class names that would otherwise
+      # conflict with types defined within the same proto file, see
+      # https://github.com/envoyproxy/envoy/pull/13378.
+      options.java_outer_classname += "OuterClass"
```

Comment:
I chose `OuterClass` because I think it is how protoc handles this case if you don't set the java_outer_classname option https://developers.google.com/protocol-buffers/docs/reference/java-generated

With this protoprint change, we don't actually _need_ the check_format error. WDYT @htuch ?

Reply:
Yeah, I agree, the check format isn't needed. I kind of don't like the special case, but my working assumption is if we change _every_ Java outer classname, this will force a lot of churn in folks who are consuming the APIs. Does that sound right? If so, I think this is the best we can do for now. Maybe leave a comment saying "TODO: In next major version, make this consistent.".

Reply:
> Does that sound right?

That's my understanding as well. All import statements and/or fully qualified class names would need to updated. I'll leave the comment.